Public/Src/Utilities/KeyValueStore/RocksDb/RocksDbStore.cs

```diff
@@ -172,8 +172,17 @@ private struct DefaultOptions
                     // The background compaction threads run in low priority, so they should not hamper the rest of
                     // the system. The number of cores in the system is what we want here according to official docs,
                     // and we are setting this to the number of logical processors, which may be higher.
+#if !PLATFORM_OSX
                     .SetMaxBackgroundCompactions(Environment.ProcessorCount)
                     .SetMaxBackgroundFlushes(1)
+#else
+                    // The memtable uses significant chunks of available system memory on macOS, we increase the number

```

Comment:
All this is probably true for Windows too, which is why it would be better if this was configurable (e.g., via some sort of RocksDB config file) instead of hardcoding it for the two platforms.  (of course, this can be done in a separate PR)